pkg/mysqlcluster/orc_reconciliation.go

```diff
@@ -72,8 +74,101 @@ func (f *cFactory) SyncOrchestratorStatus(ctx context.Context) error {
 	return nil
 }
 
+func getInstance(hostname string, insts []orc.Instance) (*orc.Instance, error) {
+
+	for _, node := range insts {
+		host := node.Key.Hostname
+
+		if host == hostname {
+			return &node, nil
+		}
+	}
+
+	return nil, fmt.Errorf("the element was not found")
+}
+
+func getMaster(node *orc.Instance, insts []orc.Instance) (string, error) {
+
+	if len(node.MasterKey.Hostname) != 0 && node.IsCoMaster == false {
+		next, err := getInstance(node.MasterKey.Hostname, insts)
+		if err == nil {
+			return getMaster(next, insts)
+		} else {
+			return "", err
+		}
+	}
+
+	if node.IsCoMaster == true {
+		next, err := getInstance(node.MasterKey.Hostname, insts)
+		if err == nil {
+			return next.Key.Hostname, nil
+		} else {
+			return "", err
+		}
+	}
+
+	return node.Key.Hostname, nil
+}
+
+func determineMasterFor(insts []orc.Instance) (string, error) {
+
+	var masterForNode []string
+
+	for _, node := range insts {
+		master, err := getMaster(&node, insts)
+		if err == nil {
+			masterForNode = append(masterForNode, master)
+		} else {
+			return "", fmt.Errorf("not able to retrieve the root of this node %s", node.Key.Hostname)
+		}
+	}
+
+	if len(masterForNode) != 0 {
+		masterHostName := masterForNode[0]
+		var check bool = true
+		for _, node := range masterForNode {
+			if node != masterHostName {
+				check = false
+			}
+
+		}
+		if check == true {
+			return masterHostName, nil
+		} else {
+			return "", fmt.Errorf("multiple masters")
+		}
+	} else {
+		return "", fmt.Errorf("0 elements in instance array")
+	}
+
+}
+
+func (f *cFactory) setMasterWritable(masterHostName string, insts []orc.Instance) error {
+	master, err := getInstance(masterHostName, insts)
```

Comment:
Get master here with `determineMasterFor`.

Also in this function you have to make sure that only one node is writable!

Reply:
Make sure that failover works as expected, by manually testing.

Test case:
 - create a cluster 2 nodes
 - delete pod 0
 - check node:0 -> read only and node:1 -> writable (cluster status)

Reply:
Test case 2:
 - create cluster 3 nodes
 - make node:1 comaster
 - check status